internal/provider/scim_group_datasource.go

```diff
@@ -0,0 +1,143 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/temporalio/terraform-provider-temporalcloud/internal/client"
+	cloudservicev1 "go.temporal.io/cloud-sdk/api/cloudservice/v1"
+	"go.temporal.io/cloud-sdk/api/identity/v1"
+)
+
+var (
+	_ datasource.DataSource              = &scimGroupDataSource{}
+	_ datasource.DataSourceWithConfigure = &scimGroupDataSource{}
+)
+
+func NewSCIMGroupDataSource() datasource.DataSource {
+	return &scimGroupDataSource{}
+}
+
+type scimGroupDataModel struct {
+	ID    types.String `tfsdk:"id"`
+	Name  types.String `tfsdk:"name"`
+	IdpId types.String `tfsdk:"idp_id"`
+}
+
+type scimGroupDataSource struct {
+	client *client.Client
+}
+
+func (d *scimGroupDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *scimGroupDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_scim_group"
+}
+
+func (d *scimGroupDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Fetches details about a SCIM group.",
+		Attributes:  scimGroupDataSourceSchema(),
+	}
+}
+
+func (d *scimGroupDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var input scimGroupDataModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &input)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Need to get group by idp ID
+	// Get user groups, filter by idp ID
+	groupResp, err := d.client.CloudService().GetUserGroups(ctx, &cloudservicev1.GetUserGroupsRequest{
```

Comment:
a noob question from me: how does client versioning work with this?

Reply:
Presumably we move this to whatever version we need to support terraform.